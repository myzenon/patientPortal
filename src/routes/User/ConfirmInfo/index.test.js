import React from 'react'
import { ConfirmInfo } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const props = {
        user: {...mockObj.patient},
        patient: {...mockObj.patient},
        page: {"data":{"chapter1Time":Date.now()}},
        savePageData: () => {
        }
    }
      const wrapper = render (
          <ConfirmInfo {...props}/>
      )
      expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  test('phone entry box will be auto format as xxx-xxx-xxxx', done => {
    const props = {
        user: {...mockObj.patient},
        patient: {...mockObj.patient},
        page: {"data":{"chapter1Time":Date.now()}},
        savePageData: () => {
        }
    }
    const wrapper = mount (
        <ConfirmInfo {...props} />
    )
    setTimeout(() => {
        wrapper.find('input[name="phoneNumber"]').simulate('change', {
            target: {
                name: 'phoneNumber',
                value: '0123456789'
            }
        })
        wrapper.find('input[name="phoneNumber"]').simulate('blur')
        expect(wrapper.state().phoneNumber).toBe('012-345-6789')
        done()
    }, 0)
})

test('validation is done when entry box loses keyboard focus', done => {
    const props = {
        user: {...mockObj.patient},
        patient: {...mockObj.patient},
        page: {"data":{"chapter1Time":Date.now()}},
        savePageData: () => {
        }
    }
    const wrapper = mount (
        <ConfirmInfo {...props} />
    )
    setTimeout(() => {
        wrapper.find('input[name="address1"]').simulate('change', {
            target: {
                name: 'address1',
                value: ''
            }
        })
        wrapper.find('input[name="address1"]').simulate('blur')
        wrapper.find('input[name="city"]').simulate('change', {
            target: {
                name: 'city',
                value: ''
            }
        })
        wrapper.find('input[name="city"]').simulate('blur')
        wrapper.find('input[name="zipCode"]').simulate('change', {
            target: {
                name: 'zipCode',
                value: ''
            }
        })
        wrapper.find('input[name="zipCode"]').simulate('blur')
        wrapper.find('input[name="phoneNumber"]').simulate('change', {
            target: {
                name: 'phoneNumber',
                value: 'abc'
            }
        })
        wrapper.find('input[name="phoneNumber"]').simulate('blur')
        wrapper.find('input[name="email"]').simulate('change', {
            target: {
                name: 'email',
                value: 'abc'
            }
        })
        wrapper.find('input[name="email"]').simulate('blur')
        expect(toJson(wrapper.render())).toMatchSnapshot()
        done()
    }, 0)
})

test('validation error | button disabled', done => {
    const props = {
        user: {...mockObj.patient},
        patient: {...mockObj.patient},
        page: {"data":{"chapter1Time":Date.now()}},
        savePageData: () => {
        }
    }
    const wrapper = mount (
        <ConfirmInfo {...props} />
    )
    setTimeout(() => {
        wrapper.find('input[name="address1"]').simulate('change', {
            target: {
                name: 'address1',
                value: ''
            }
        })
        wrapper.find('input[name="address1"]').simulate('blur')
        expect(toJson(wrapper.find('button[type="submit"]').render())).toMatchSnapshot()
        done()
    }, 0)
})

test('update address1 | clicks next button | api fhirPatientsIdPut called | previousConfirmedInfo === true |  go to appropriate page', done => {
    const props = {
        user: {...mockObj.patient},
        patient: {...mockObj.patient},
        page: {"data":{"chapter1Time":Date.now()}},
        savePageData: () => {
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirPatientsIdPut')
            expect(get).toEqual({
                id: mockObj.patient.id
            })
            expect(post).toEqual({
                ...mockObj.patient,
                address: [
                    {
                        ...mockObj.patient.address[0],
                        line: [
                            'My Home'
                        ]
                    }
                ],
                confirmedInfo: true
            })
            return Promise.resolve()
        },
        setPatient: (patient) => {
            expect(patient).toEqual({
                ...mockObj.patient,
                address: [
                    {
                        ...mockObj.patient.address[0],
                        line: [
                            'My Home'
                        ]
                    }
                ],
                confirmedInfo: true
            })
            return Promise.resolve()
        },
        goToPageAfterAuth: () => {
            done()
            return Promise.resolve()
        }
    }
    const wrapper = mount (
        <ConfirmInfo {...props} />
    )
    setTimeout(() => {
        wrapper.find('input[name="address1"]').simulate('change', {
            target: {
                name: 'address1',
                value: 'My Home'
            }
        })
        wrapper.find('input[name="address1"]').simulate('blur')
        wrapper.find('button[type="submit"]').simulate('submit')
    }, 0)
})

test('clicks next button | confirmedInfo === false |  go to pcp query', done => {
    const props = {
        user: {
            ...mockObj.patient,
            confirmedInfo: false
        },
        patient: {
            ...mockObj.patient,
            confirmedInfo: false
        },
        page: {"data":{"chapter1Time":Date.now()}},
        savePageData: () => {
        },
        callAPIGateway: () => Promise.resolve(),
        setPatient: () => Promise.resolve(),
        push: (url) => {
            expect(url).toBe('/pcp-query')
            done()
        }
    }
    const wrapper = mount (
        <ConfirmInfo {...props} />
    )
    setTimeout(() => {
        wrapper.find('input[name="address1"]').simulate('change', {
            target: {
                name: 'address1',
                value: 'My Home'
            }
        })
        wrapper.find('input[name="address1"]').simulate('blur')
        wrapper.find('button[type="submit"]').simulate('submit')
    }, 0)
})
