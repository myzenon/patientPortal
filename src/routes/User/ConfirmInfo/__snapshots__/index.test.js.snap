// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <form
      class=""
    >
      <div
        class="inputWrapper row"
      >
        <div
          class="col-12 col-md-7 offset-md-2"
        >
          <span
            class="heading"
          >
            Please review your contact information and confirm that it is accurate and up to date. You can click to edit any field.
            <p />
          </span>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-3 offset-md-2"
        >
          <strong>
            Full Name
          </strong>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formText col-12 col-md-3 offset-md-2"
        >
          <strong>
            Rita Summers
          </strong>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-4 offset-md-2"
        >
          <label>
            Address 1
          </label>
          <div
            class="addresstextboxwrapper"
          >
            <input
              autofocus=""
              class="inputText"
              name="address1"
              type="text"
              value="47 Inverness Road"
            />
          </div>
        </div>
        <div
          class="formLabel col-12 col-md-3"
        >
          <label>
            Address 2
          </label>
          <div
            class="addresstextboxwrapper1"
          >
            <input
              class="inputText"
              name="address2"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-4 offset-md-2"
        >
          <label>
            City
          </label>
          <div
            class="addresstextboxwrapper"
          >
            <input
              class="inputText"
              name="city"
              type="text"
              value="Gastonia"
            />
          </div>
        </div>
        <div
          class="formLabel col-12 col-md-3"
        >
          <label>
            State
          </label>
          <div
            class="addresstextboxwrapper1"
          >
            <span
              class="selectLabel"
            >
              <select
                class="selectLeft"
                isleftalign="true"
                name="state"
                type="text"
              >
                <option
                  value="AL"
                >
                  Alabama
                </option>
                <option
                  value="AK"
                >
                  Alaska
                </option>
                <option
                  value="AZ"
                >
                  Arizona
                </option>
                <option
                  value="AR"
                >
                  Arkansas
                </option>
                <option
                  value="CA"
                >
                  California
                </option>
                <option
                  value="CO"
                >
                  Colorado
                </option>
                <option
                  value="CT"
                >
                  Connecticut
                </option>
                <option
                  value="DE"
                >
                  Delaware
                </option>
                <option
                  value="FL"
                >
                  Florida
                </option>
                <option
                  value="GA"
                >
                  Georgia
                </option>
                <option
                  value="HI"
                >
                  Hawaii
                </option>
                <option
                  value="ID"
                >
                  Idaho
                </option>
                <option
                  value="IL"
                >
                  Illinois
                </option>
                <option
                  value="IN"
                >
                  Indiana
                </option>
                <option
                  value="IA"
                >
                  Iowa
                </option>
                <option
                  value="KS"
                >
                  Kansas
                </option>
                <option
                  value="KY"
                >
                  Kentucky
                </option>
                <option
                  value="LA"
                >
                  Louisiana
                </option>
                <option
                  value="ME"
                >
                  Maine
                </option>
                <option
                  value="MD"
                >
                  Maryland
                </option>
                <option
                  value="MA"
                >
                  Massachusetts
                </option>
                <option
                  value="MI"
                >
                  Michigan
                </option>
                <option
                  value="MN"
                >
                  Minnesota
                </option>
                <option
                  value="MS"
                >
                  Mississippi
                </option>
                <option
                  value="MO"
                >
                  Missouri
                </option>
                <option
                  value="MT"
                >
                  Montana
                </option>
                <option
                  value="NE"
                >
                  Nebraska
                </option>
                <option
                  value="NV"
                >
                  Nevada
                </option>
                <option
                  value="NH"
                >
                  New Hampshire
                </option>
                <option
                  value="NJ"
                >
                  New Jersey
                </option>
                <option
                  value="NM"
                >
                  New Mexico
                </option>
                <option
                  value="NY"
                >
                  New York
                </option>
                <option
                  selected=""
                  value="NC"
                >
                  North Carolina
                </option>
                <option
                  value="ND"
                >
                  North Dakota
                </option>
                <option
                  value="OH"
                >
                  Ohio
                </option>
                <option
                  value="OK"
                >
                  Oklahoma
                </option>
                <option
                  value="OR"
                >
                  Oregon
                </option>
                <option
                  value="PA"
                >
                  Pennsylvania
                </option>
                <option
                  value="RI"
                >
                  Rhode Island
                </option>
                <option
                  value="SC"
                >
                  South Carolina
                </option>
                <option
                  value="SD"
                >
                  South Dakota
                </option>
                <option
                  value="TN"
                >
                  Tennessee
                </option>
                <option
                  value="TX"
                >
                  Texas
                </option>
                <option
                  value="UT"
                >
                  Utah
                </option>
                <option
                  value="VT"
                >
                  Vermont
                </option>
                <option
                  value="VA"
                >
                  Virginia
                </option>
                <option
                  value="WA"
                >
                  Washington
                </option>
                <option
                  value="WV"
                >
                  West Virginia
                </option>
                <option
                  value="WI"
                >
                  Wisconsin
                </option>
                <option
                  value="WY"
                >
                  Wyoming
                </option>
              </select>
            </span>
          </div>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-4 offset-md-2"
        >
          <label>
            Phone Number
          </label>
          <div
            class="addresstextboxwrapper"
          >
            <input
              class="inputText"
              name="phoneNumber"
              placeholder="XXX-XXX-XXXX"
              type="text"
              value="732-222-1333"
            />
          </div>
        </div>
        <div
          class="formLabel col-12 col-md-3"
        >
          <label>
            Zip Code
          </label>
          <div
            class="addresstextboxwrapper1"
          >
            <input
              class="inputText"
              name="zipCode"
              type="text"
              value="28052"
            />
          </div>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="emailformLabel col-12 col-md-7 offset-md-2"
        >
          <label>
            The pharmacist should email my recommendations to
          </label>
          <input
            class="inputText"
            name="email"
            type="email"
            value=""
          />
        </div>
        <p />
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-7 offset-md-2"
        >
          <div
            class="heading3"
          >
            Once our pharmacist completes your medication review, we will email a link of your personal medication list and medication action plan.
          </div>
        </div>
      </div>
      <div
        class="selectCheckWrapper row"
      >
        <div
          class="checkboxWrapper col-12 col-md-7 offset-md-2"
        >
          <label
            class="chkContainer"
          >
            <input
              name="getByEmail"
              type="checkbox"
            />
            <span
              class="checkbox"
            />
          </label>
          <span
            class="checkboxLabelWrapper"
          >
            In addition to the email, I would like a hard copy mailed to me.
          </span>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="buttonWrapper col-12 col-md-4 offset-md-3"
        >
          <button
            class="button btn btn-secondary"
            id="btnNextConfirmInfo"
            type="submit"
          >
            <b>
              NEXT
            </b>
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepActive"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="step"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumber"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`validation error | button disabled 1`] = `
<button
  class="button btn btn-secondary"
  id="btnNextConfirmInfo"
  type="submit"
>
  <b>
    NEXT
  </b>
</button>
`;

exports[`validation is done when entry box loses keyboard focus 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <form
      class=""
    >
      <div
        class="inputWrapper row"
      >
        <div
          class="col-12 col-md-7 offset-md-2"
        >
          <span
            class="heading"
          >
            Please review your contact information and confirm that it is accurate and up to date. You can click to edit any field.
            <p />
          </span>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-3 offset-md-2"
        >
          <strong>
            Full Name
          </strong>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formText col-12 col-md-3 offset-md-2"
        >
          <strong>
            Rita Summers
          </strong>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-4 offset-md-2"
        >
          <label>
            Address 1
          </label>
          <div
            class="addresstextboxwrapper"
          >
            <input
              class="inputTextError"
              name="address1"
              type="text"
              value=""
            />
          </div>
          <div
            class="textError"
          >
            Address is Required.
          </div>
        </div>
        <div
          class="formLabel col-12 col-md-3"
        >
          <label>
            Address 2
          </label>
          <div
            class="addresstextboxwrapper1"
          >
            <input
              class="inputText"
              name="address2"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-4 offset-md-2"
        >
          <label>
            City
          </label>
          <div
            class="addresstextboxwrapper"
          >
            <input
              class="inputTextError"
              name="city"
              type="text"
              value=""
            />
          </div>
          <div
            class="textError"
          >
            City is Required.
          </div>
        </div>
        <div
          class="formLabel col-12 col-md-3"
        >
          <label>
            State
          </label>
          <div
            class="addresstextboxwrapper1"
          >
            <span
              class="selectLabel"
            >
              <select
                class="selectLeft"
                isleftalign="true"
                name="state"
                type="text"
              >
                <option
                  value="AL"
                >
                  Alabama
                </option>
                <option
                  value="AK"
                >
                  Alaska
                </option>
                <option
                  value="AZ"
                >
                  Arizona
                </option>
                <option
                  value="AR"
                >
                  Arkansas
                </option>
                <option
                  value="CA"
                >
                  California
                </option>
                <option
                  value="CO"
                >
                  Colorado
                </option>
                <option
                  value="CT"
                >
                  Connecticut
                </option>
                <option
                  value="DE"
                >
                  Delaware
                </option>
                <option
                  value="FL"
                >
                  Florida
                </option>
                <option
                  value="GA"
                >
                  Georgia
                </option>
                <option
                  value="HI"
                >
                  Hawaii
                </option>
                <option
                  value="ID"
                >
                  Idaho
                </option>
                <option
                  value="IL"
                >
                  Illinois
                </option>
                <option
                  value="IN"
                >
                  Indiana
                </option>
                <option
                  value="IA"
                >
                  Iowa
                </option>
                <option
                  value="KS"
                >
                  Kansas
                </option>
                <option
                  value="KY"
                >
                  Kentucky
                </option>
                <option
                  value="LA"
                >
                  Louisiana
                </option>
                <option
                  value="ME"
                >
                  Maine
                </option>
                <option
                  value="MD"
                >
                  Maryland
                </option>
                <option
                  value="MA"
                >
                  Massachusetts
                </option>
                <option
                  value="MI"
                >
                  Michigan
                </option>
                <option
                  value="MN"
                >
                  Minnesota
                </option>
                <option
                  value="MS"
                >
                  Mississippi
                </option>
                <option
                  value="MO"
                >
                  Missouri
                </option>
                <option
                  value="MT"
                >
                  Montana
                </option>
                <option
                  value="NE"
                >
                  Nebraska
                </option>
                <option
                  value="NV"
                >
                  Nevada
                </option>
                <option
                  value="NH"
                >
                  New Hampshire
                </option>
                <option
                  value="NJ"
                >
                  New Jersey
                </option>
                <option
                  value="NM"
                >
                  New Mexico
                </option>
                <option
                  value="NY"
                >
                  New York
                </option>
                <option
                  value="NC"
                >
                  North Carolina
                </option>
                <option
                  value="ND"
                >
                  North Dakota
                </option>
                <option
                  value="OH"
                >
                  Ohio
                </option>
                <option
                  value="OK"
                >
                  Oklahoma
                </option>
                <option
                  value="OR"
                >
                  Oregon
                </option>
                <option
                  value="PA"
                >
                  Pennsylvania
                </option>
                <option
                  value="RI"
                >
                  Rhode Island
                </option>
                <option
                  value="SC"
                >
                  South Carolina
                </option>
                <option
                  value="SD"
                >
                  South Dakota
                </option>
                <option
                  value="TN"
                >
                  Tennessee
                </option>
                <option
                  value="TX"
                >
                  Texas
                </option>
                <option
                  value="UT"
                >
                  Utah
                </option>
                <option
                  value="VT"
                >
                  Vermont
                </option>
                <option
                  value="VA"
                >
                  Virginia
                </option>
                <option
                  value="WA"
                >
                  Washington
                </option>
                <option
                  value="WV"
                >
                  West Virginia
                </option>
                <option
                  value="WI"
                >
                  Wisconsin
                </option>
                <option
                  value="WY"
                >
                  Wyoming
                </option>
              </select>
            </span>
          </div>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-4 offset-md-2"
        >
          <label>
            Phone Number
          </label>
          <div
            class="addresstextboxwrapper"
          >
            <input
              class="inputTextError"
              name="phoneNumber"
              placeholder="XXX-XXX-XXXX"
              type="text"
              value="abc"
            />
          </div>
          <div
            class="textError"
          >
            Use only numbers.
          </div>
        </div>
        <div
          class="formLabel col-12 col-md-3"
        >
          <label>
            Zip Code
          </label>
          <div
            class="addresstextboxwrapper1"
          >
            <input
              class="inputTextError"
              name="zipCode"
              type="text"
              value=""
            />
          </div>
          <div
            class="textError"
          >
            ZipCode is Required.
          </div>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="emailformLabel col-12 col-md-7 offset-md-2"
        >
          <label>
            The pharmacist should email my recommendations to
          </label>
          <input
            class="inputTextError"
            name="email"
            type="email"
            value="abc"
          />
          <div
            class="textError"
          >
            A valid email is Required.
          </div>
        </div>
        <p />
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="formLabel col-12 col-md-7 offset-md-2"
        >
          <div
            class="heading3"
          >
            Once our pharmacist completes your medication review, we will email a link of your personal medication list and medication action plan.
          </div>
        </div>
      </div>
      <div
        class="selectCheckWrapper row"
      >
        <div
          class="checkboxWrapper col-12 col-md-7 offset-md-2"
        >
          <label
            class="chkContainer"
          >
            <input
              name="getByEmail"
              type="checkbox"
            />
            <span
              class="checkbox"
            />
          </label>
          <span
            class="checkboxLabelWrapper"
          >
            In addition to the email, I would like a hard copy mailed to me.
          </span>
        </div>
      </div>
      <div
        class="inputWrapper row"
      >
        <div
          class="buttonWrapper col-12 col-md-4 offset-md-3"
        >
          <button
            class="button btn btn-secondary"
            id="btnNextConfirmInfo"
            type="submit"
          >
            <b>
              NEXT
            </b>
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepActive"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="step"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumber"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
